client/src/components/Auth/AuthProvider.js

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from 'react'
+import axios from 'axios'
+import jwt from 'jwt-decode'
+
+const AuthContext = React.createContext(null)
+
+const AuthProvider = ({ children }) => {
+    const [auth, setAuth] = useState(null)
+
+    const login = async (email, password, callback) => { 
+        console.log("[Login]")
+        try{
+            const authResponse = await axios.post(
+                'http://localhost:8000/auth/login', 
+                { email: email, password: password }, 
+                { 'content-type': 'application/json' }
+            )
+            const decoded = jwt(authResponse.data.token)
+            setAuth({ token: authResponse.data.token, user: decoded.user })
+            callback()
+        } catch (err) {
+            console.log(`Login error ${err}`)
+            alert('Email and/or password not recognized')
+            // Assignment: what should we do if this fails?
+        }
+    }
+
+    const register = async (email, password, callback) => { 
+        // Assignment: how do we register someone? 
+        console.log("[Register]")
+        try{
+            const registerResponse = await axios.post(
+                'http://localhost:8000/auth/register', 
+                { email: email, password: password }, 
+                { 'content-type': 'application/json' }
+            )
+            const decoded = jwt(registerResponse.data.token)
+            setAuth({ token: registerResponse.data.token, user: decoded.user })
```

Comment:
Do you really expect to get a token back? Should you?

Reply:
no token expected, i forgot to edit this out.